onnxruntime/core/providers/coreml/builders/impl/builder_utils.cc

```diff
@@ -290,6 +298,12 @@ MILSpec::Value CreateScalarTensorValue(const T& data) {
 // explicit specializations for types we handle so the implementation can be in the .cc file
 template MILSpec::Value CreateTensorValue<int64_t, int32_t>(gsl::span<const int64_t> data,
                                                             std::optional<gsl::span<const int64_t>> shape);
+template MILSpec::Value CreateTensorValue<float, float>(gsl::span<const float> data,
+                                                        std::optional<gsl::span<const int64_t>> shape);
+template MILSpec::Value CreateTensorValue<bool, bool>(gsl::span<const bool> data,
+                                                      std::optional<gsl::span<const int64_t>> shape);
+template MILSpec::Value CreateTensorValue<std::string, std::string>(gsl::span<const std::string> data,
+                                                                    std::optional<gsl::span<const int64_t>> shape);
```

Comment:
Why do we need these new specializations to handle fp16 data?

Reply:
The Linux-GCC compiler complains that ` CreateTensorValue<float, float>` is not defined.  The template instantiation helps to suppression it.

Reply:
Yes, It's a forward declaration for specific function signature.